src/emsdevice.cpp

```diff
@@ -753,20 +754,24 @@ bool EMSdevice::generate_values_json(JsonObject & root, const uint8_t tag_filter
                 }
             }
         }
+        dv.ha      |= has_value ? DeviceValueHA::HA_VALUE : DeviceValueHA::HA_NONE;
+        has_values |= has_value;
     }
 
-    return has_value;
+    return has_values;
 }
 
 // create the Home Assistant configs for each value
-// this is called when an MQTT publish is done via an EMS Device, and only done once
+// this is called when an MQTT publish is done via an EMS Device
 void EMSdevice::publish_mqtt_ha_sensor() {
-    for (const auto & dv : devicevalues_) {
-        Mqtt::publish_mqtt_ha_sensor(dv.type, dv.tag, dv.full_name, device_type_, dv.short_name, dv.uom);
+    for (auto & dv : devicevalues_) {
+        if (dv.ha == DeviceValueHA::HA_VALUE) {
+            Mqtt::publish_mqtt_ha_sensor(dv.type, dv.tag, dv.full_name, device_type_, dv.short_name, dv.uom);
+            dv.ha |= DeviceValueHA::HA_DONE;
+        }
     }
-
-    bool ok = publish_ha_config();
-    ha_config_done(ok); // see if it worked
+    // bool ok = publish_ha_config();
```

Comment:
this command is still needed for Home Assistant to build the main device (then all the sensors attach to it). When uncommented it works great!

Also maybe look at ha_config_done() as it really doesn't do anything. The idea was if you change the Home Assistant switch on the Web UI it would re-send all the Discovery HA's again.

Why don't you install Home Assistant yourself? It's really nice software. B-)

Reply:
Yes i've seen it and commited a fix a few minutes ago.